src/JSConfetti.ts

```diff
@@ -4,17 +4,59 @@ import { createCanvas } from './createCanvas'
 import { normalizeConfettiConfig } from './normalizeConfettiConfig'
 import { IPosition, IJSConfettiConfig, IAddConfettiConfig } from './types'
 
+class ConfettiBatch {
+  private promiseCompletion?: () => void
+  private promise?: Promise<void>
+  private members: ConfettiShape[]
```

Comment:
let's call them `shapes` as they were called in `JSConfetti` class?

Reply:
Sure!